packages/beacon-node/src/chain/validation/block.ts

```diff
@@ -111,13 +111,13 @@ export async function validateGossipBlock(
     });
   }
 
-  // getBlockSlotState also checks for whether the current finalized checkpoint is an ancestor of the block.
+  // use getPreState to reload state if needed. It also checks for whether the current finalized checkpoint is an ancestor of the block.
   // As a result, we throw an IGNORE (whereas the spec says we should REJECT for this scenario).
   // this is something we should change this in the future to make the code airtight to the spec.
   // [IGNORE] The block's parent (defined by block.parent_root) has been seen (via both gossip and non-gossip sources) (a client MAY queue blocks for processing once the parent block is retrieved).
   // [REJECT] The block's parent (defined by block.parent_root) passes validation.
   const blockState = await chain.regen
-    .getBlockSlotState(parentRoot, blockSlot, {dontTransferCache: true}, RegenCaller.validateGossipBlock)
+    .getPreState(block, {dontTransferCache: true}, RegenCaller.validateGossipBlock)
```

Comment:
this is the same to `getBlockSlotState()` except for it allows to reload state if needed
if block epoch is same to parent epoch, there is no need to dial to block slot
if block epoch > parent epoch, it'll use checkpoint state of block epoch. In `regen.ts` it'll use `getBlockSlotState()` in the end https://github.com/ChainSafe/lodestar/blob/822446fd155f0ff5585787dbe3d8a1dade44d269/packages/beacon-node/src/chain/regen/regen.ts#L91

Reply:
I think it's just the same, if parentEpoch < blockEpoch it'll get checkpoint state. The `PrepareNextSlot` should do the epoch transition for us anyway so epoch transition doesn't come from here